### PR TITLE
graphql: add support for `order_by` clause

### LINKED ISSUE #13: graphql: add support for `order_by` clause
Some of the options we could potentially support.

image

### PR DESCRIPTION
**Pull request by kolharsam**
the PR should mainly resolve #13

### CODE REVIEW

**Comment by kolharsam** [comment]
This marks the end of stage - 1. To get the basic functionality working. Now, refactor! :)

File: src/types.rs [hunk 1]
```diff
@@ -162,23 +163,33 @@ fn field_name_to_sql_with_alias() {
...
+// functions for the processing of the AST and SQL code
+pub enum GQLArgType<T> {
     // NOTE: supported for [distinct_on]
     String(String),
     // NOTE: supported for [limit, offset]
     Int(i64),
-    // Float(f32)
+    // NOTE: supported for [order_by]
+    Object(IndexMap<String, T>),
 }
```
Comment (iykekings) [inline src/types.rs:178]: I'm a bit worried that this enum field doesn't clearly show what they are for. can we consider sth like this?
```rust
pub enum GQLArgType<T> {
    // NOTE: supported for [distinct_on]
    DistinctOn(String),
    // NOTE: supported for [limit, offset]
    Limit(i64),
    Offset(i64),
    // NOTE: supported for [order_by]
    OrderBy(IndexMap<String, T>),
}
```

**Review by iykekings — APPROVED** [review]
Code looks super good and functionality works!

File: src/types.rs (hunk 1 above)
Comment (kolharsam) [reply src/types.rs:178]: I am in favour of this, but this refactor can be picked up at a later stage, I presume? Keeping it argument agnostic, I think we can re-use a lot more code. Plus, there's only one more `argtype` that will be added for queries, i.e `where`.

Comment (kolharsam) [reply src/types.rs:178]: I'll be merging this PR now, but yeah, you could explain your reasoning too and we can take a more informed decision upon that.

**Pull request merged**